src/market.py

```diff
@@ -143,18 +148,60 @@ class OrderBook:
     slab: Slab
 
     def __init__(self, market: Market, account_flags: Any, slab: Slab):
+        if not account_flags.initialized or not account_flags.bids ^ account_flags.asks:
+            raise Exception("Invalid order book, either not initialized or neither of bids or asks")
         self.market = market
-        self.is_bids = account_flags
+        self.is_bids = account_flags.bids
         self.slab = slab
 
     @staticmethod
-    def decode(market: Market, buffer):
+    def decode(market: Market, buffer: bytes):
         """Decode the given buffer into an order book."""
-        raise NotImplementedError("decode is not implemented yet")
+        # This is a bit hacky at the moment. The first 5 bytes are padding, the
+        # total length is 8 bytes which is 5 + 8 = 13 bytes.
+        account_flags = ACCOUNT_FLAGS_LAYOUT.parse(buffer[5:13])
+        slab = Slab.decode(buffer[13:])
+        return OrderBook(market, account_flags, slab)
 
-    def get_l2(self, depth: int):
+    def get_l2(self, depth: int) -> List[Tuple[float, float, int, int]]:
         """Get the Level 2 market information."""
-        raise NotImplementedError("get_l2 is not implemented yet")
+        descending = self.is_bids
+        levels: List[List[int]] = []
```

Comment:
Can we use a named tuple here instead of `List[int]` for `[price, node.quantity]`.

Reply:
Don't think it will be a good idea in this case since we need to modify the content.

Reply:
Can you leave a comment here that 0 indexes to price and 1 indexes to amount/quantity.

Reply:
Added comment.